File: api/SecurityMonitoringAI.API/SecurityMonitoringAI.API/Models/SecurityEvent.cs
namespace SecurityMonitoring.Api.Models;

public class SecurityEvent
{
    public int Id { get; set; }

    public DateTime Timestamp { get; set; }

    public string SourceType { get; set; } = string.Empty;

    public string SourceIp { get; set; } = string.Empty;

    public string? DestinationIp { get; set; }

    public int? SourcePort { get; set; }

    public int? DestinationPort { get; set; }

    public string? Protocol { get; set; }

    public int? EventId { get; set; }

    public string EventType { get; set; } = string.Empty;

    public string Severity { get; set; } = string.Empty;

    public string Message { get; set; } = string.Empty;

    public string Status { get; set; } = "New";
}

File: client/SecurityMonitoring.Client/src/app/core/models/security-event.ts
export interface SecurityEvent {
  id: number;
  timestamp: string;
  sourceType: string;
  sourceIp: string;
  destinationIp: string | null;
  sourcePort: number | null;
  destinationPort: number | null;
  protocol: string | null;
  eventId: number | null;
  eventType: string;
  severity: string;
  message: string;
  status: string;
}